distribution/lib/Standard/Base/0.0.0-dev/src/Data/Vector.enso

```diff
@@ -723,7 +723,7 @@ type Vector a
 
              [1] + [2]
     + : Vector Any -> Vector Any
-    + self that = case that of
+    + self that:Vector = case that of
         _ : Vector -> Vector.insert_builtin self self.length that
         _ : Array -> self + Vector.from_polyglot_array that
         _ -> Error.throw (Type_Error.Error Vector that "that")
```

Comment:
Hmmm, something seems off here now.

Will `vector + array` still work? I assume it won't, beacuse in the REPL I get:
```
> [12,3].to_array.to Vector
Evaluation failed with: Could not find a conversion from `[12, 3]` to `Vector`.
java.lang.Exception: Could not find a conversion from `[12, 3]` to `Vector`.
```
unless there is some irregular special handling for the Array/Vector case in the typecheck node.